cmd/internal/datastore/s3/transaction.go

```diff
@@ -0,0 +1,55 @@
+package s3
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/mailchain/mailchain/encoding"
+
+	"github.com/mailchain/mailchain/stores/s3store"
+)
+
+type TransactionStore struct {
+	*s3store.Uploader
```

Comment:
Try to use composition over inheritance. `uploader = *s3store.Uploader` is preferred rather than`TransactionStore` inheriting from `s3store.Uploader`